src/unix/udp.c

```diff
@@ -660,6 +754,35 @@ int uv_udp_set_membership(uv_udp_t* handle,
   }
 }
 
+
+int uv_udp_set_source_membership(uv_udp_t* handle,
+                                 const char* multicast_addr,
+                                 const char* interface_addr,
+                                 const char* source_addr,
+                                 uv_membership membership) {
+  int err;
+  struct sockaddr_in mcast_addr4;
+  struct sockaddr_in src_addr4;
+  struct sockaddr_in6 mcast_addr6;
+  struct sockaddr_in6 src_addr6;
+
+  err = uv_ip4_addr(multicast_addr, 0, &mcast_addr4);
+  if (err) {
+    err = uv_ip6_addr(multicast_addr, 0, &mcast_addr6);
+    if (err)
+      return err;
+    err = uv_ip6_addr(source_addr, 0, &src_addr6);
+    if (err)
+      return err;
+    return uv__udp_set_source_membership6(handle, &mcast_addr6, interface_addr, &src_addr6, membership);
+  }
+  err = uv_ip4_addr(source_addr, 0, &src_addr4);
+  if (err)
+    return err;
+  return uv__udp_set_source_membership4(handle, &mcast_addr4, interface_addr, &src_addr4, membership);
```

Comment:
Style: long lines. Max should be 80 characters.

Reply:
fixed